container/go/cmd/pusher/pusher.go

```diff
@@ -126,8 +127,17 @@ func main() {
 		log.Printf("Failed to digest image: %v", err)
 	}
 
-	if err := push(stamped, img); err != nil {
-		log.Fatalf("Error pushing image to %s: %v", stamped, err)
+	for retry := 0; retry < *retryCount+1; retry++ {
+		err := push(stamped, img)
+		if err == nil {
+			break
+		}
+
+		if *retryCount > 0 && retry < *retryCount {
+			log.Printf("Failed to push image on attempt %d: %v", retry+1, err)
+		} else {
+			log.Fatalf("Error pushing image to %s: %v", stamped, err)
+		}
```

Comment:
Thanks for taking a look! The reason I implemented it this way is that the actual issue we have is that sometimes it fails to load the auth credentials for GCR for 1 or 2 pushes out of > 30. This way it covers that scenario as well and means we don't have to rerun the entire build.

Reply:
I think this is reasonable. It would be cleaner to use @lbcjbb's suggestion, but I also empathise with flakiness beyond just pushing to container registries.

Reply:
After verification `WithRetry*` options are not available in `v0.5.1` of the library, so we can't use them for the moment.

Reply:
Yea, that definitely feels like a nicer solution if it was only network issues that caused the problems.